apps/passport-client/src/localstorage.ts

```diff
@@ -31,10 +42,18 @@ export async function loadPCDs(): Promise<PCDCollection> {
   }
 
   const serializedCollection = window.localStorage[COLLECTION_KEY];
-  return await PCDCollection.deserialize(
+  const collection = await PCDCollection.deserialize(
     await getPackages(),
     serializedCollection ?? "{}"
   );
+
+  const validationErrors = validatePCDCollection(collection);
+  if (validationErrors.length > 0) {
+    logAndUploadValidationErrors(validationErrors);
+    throw new Error("couldn't save PCDs\n:" + validationErrors.join("\n"));
```

Comment:
This is actually loading not saving (typo) and we need to think about where this error goes and whether the user can find out about it in a reasonable way.  The user probably needs to fix it by logging off, but they might not be able to if the page crashes.

Reply:
I wasn't sure about how to make sure that the right thing got communicated from this layer of the stack back to the user. Things I considered included:
- throwing up an `alert`, which can be done from any depth
- throwing a particular type of error that the rest of the application knows how to handle, similarly to how we do it in in the server with `PCDHTTPError`
- factoring out the save logic into some other place which has access to more of the application state, so that it can both set an error and reject inconsistent saves at this level

I think longer term, this question relates to #1291 (regarding which I did some research yesterday).

Reply:
Agreed, and the complexity of how to respond correctly is part of why I felt the need to ask for help with this issue. :)

There isn't any notion of a top-level error handler for the whole app, is there?  It would be nice to have one place to specify that a specific error type/name/message should be handled in a specific way.

Centralizing the load/save logic (at least for synced state) into a single global object which can have broader context (like access to AppState and/or a way to report errors) may be a good approach.  I think all the sync logic probably should eventually migrate into some sort of SyncManager to own things like that, but that's too big a refactor to do in December.